src/common/i2c.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(dead_code)]
+
+use super::config::{AttributeError, Kind};
+use std::sync::{Arc, Mutex};
+
+// A trait representing blocking I2C communication for a board. TODO: replace with the
+// embedded_hal I2C trait when supporting boards beyond ESP32. AddressType is
+// either u8 (indicating support for 7-bit addresses) or u16 (for supporting 10-bit addresses)
+pub trait I2CHandle<AddressType> {
+    fn name(&self) -> String;
+
+    fn read_i2c(&mut self, _address: AddressType, _buffer: &mut [u8]) -> anyhow::Result<()> {
+        anyhow::bail!("read_i2c unimplemented")
```

Comment:
might be worth considering adding a function like this one (https://docs.rs/embedded-hal/latest/embedded_hal/blocking/i2c/trait.WriteIterRead.html) since some I2c hardware requires you to write to a register before reading it. We can wait until we are implementing an actual hardware piece !

Reply:
funny story; I had write_read in the trait previously but I removed it by accident, I'll add it back